### PR TITLE
Feat(torii-grpc): support Events query

### LINKED ISSUE #1420: [Torii] gRPC to support Events query
Support event query and subscription over grpc, similar to graphql. 

This should be pretty straight forward to implement, very similar to how entity query/sub is implemented.

### PR DESCRIPTION
**Pull request by gianalarcon**
Closes #1420 

Hi @broody Please check my  Events query implementation. Waiting for your feedback

### CODE REVIEW

File: crates/torii/grpc/proto/types.proto [hunk 1]
```diff
@@ -84,10 +99,21 @@ message Clause {
     }
 }
 
+message EventClause {
+    oneof event_clause_type {
+        HashedKeysClause hashed_keys = 1;
```
[the remark this answers is not in the document]
Comment (gianalarcon) [reply crates/torii/grpc/proto/types.proto:104]: Thanks for your reply. Gonna work on it

File: crates/torii/grpc/proto/types.proto [hunk 2]
```diff
@@ -45,6 +45,15 @@ message Entity {
     repeated Model models = 2;
 }
 
+message Event {
+    // The event's hashed keys
+    bytes hashed_keys = 1;
```
[the remark this answers is not in the document]
Comment (gianalarcon) [reply crates/torii/grpc/proto/types.proto:50]: Actually `message Event` is used on the Response/output. Thus, the output should display `bytes hashed_keys` or `repeated bytes keys`?

File: crates/torii/grpc/proto/types.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (gianalarcon) [reply crates/torii/grpc/proto/types.proto]: Fixed on dcaa4e0

File: crates/torii/grpc/src/server/mod.rs [hunk 3]
```diff
@@ -126,6 +127,18 @@ impl DojoWorld {
         self.entities_by_hashed_keys(None, limit, offset).await
     }
 
+    async fn events_all(&self, limit: u32, offset: u32) -> Result<Vec<proto::types::Event>, Error> {
+        let query = r#"
+            SELECT keys, data, transaction_hash
+            FROM events
+            LIMIT ? OFFSET ?
```
Comment (broody) [inline crates/torii/grpc/src/server/mod.rs:134]: I think we'd want to order by events id (descending) so results are chronological

File: crates/torii/grpc/src/server/mod.rs [hunk 4]
```diff
@@ -245,6 +258,41 @@ impl DojoWorld {
...
+                Ok(str::from_utf8(bytes).unwrap().to_string())
+            })
+            .collect::<Result<Vec<_>, Error>>()?;
+        let keys_pattern = keys.join("/") + "/%";
+
+        let events_query = r#"
+            SELECT keys, data, transaction_hash
+            FROM events
+            WHERE keys LIKE ?
+            LIMIT ? OFFSET ?
```
Comment (broody) [inline crates/torii/grpc/src/server/mod.rs:283]: Ordering here as well since pattern matching on keys

**Comment by gianalarcon** [comment]
> Just ordering of results then I think we're good to go.

Fixed on commit 7e09a0c

**Review by broody — APPROVED** [review]

**Pull request merged**